OpenRA.Mods.Common/Widgets/EditorViewportControllerWidget.cs

```diff
@@ -35,7 +36,28 @@ public EditorViewportControllerWidget(World world, WorldRenderer worldRenderer)
 			CurrentBrush = defaultBrush = new EditorDefaultBrush(this, worldRenderer);
 		}
 
+		public void ToggleEyedropping()
```

Comment:
The modifications in this file aren't needed.  Just set the brush type directly from the `MapEditorLogic` (see how the copy/paste brush does it).